Food rebalance

hivehum opened the pull request:
the goal of this pr is to rebalance EVERY SINGLE CRAFTABLE FOOD ITEM IN THE GAME to _at minimum_ contain the sum of the reagents in its ingredients. im not kidding when i say that almost every food item in this game obliterates half the nutritional value of its contents. its so fucking bad in here

this pr also fixes a completely unnoticed bug that removed all nutritional content from almost every vending machine item (pr #194 overwrote the reagent inventory of them to add toxicity for snails). after consulting with rnd i have returned nutriment to these items but substantially nerfed the amount to further incentivize going to the kitchen. not like you need much more incentive because basically everything a chef can cook is much more filling than ever before

also hit nutribrick with a hammer because this was my original goal lol. soap moment really hard

additionally this resolves two upstream todos and adds blackpepper to recipe for queso and bananium to clown tears
finfin cutlets are cookable. this appears to have broken around when burger towers were added

oh yeah this started out as a port of #34138 but at this point i have completely gone back and redone all of stop-signs' work

**Changelog**
:cl:
- tweak: Nutribricks are far less nutritious...
- tweak: Chefs will notice a subtle but substantial increase in the intensity of their output.
- fix: Yeah um so it turns out every food from a vending machine had no nutritional value. That's fixed now. Oops.

hivehum commented:
well thats an extremely scary test fail that i have no idea what to do with. anyway this is ready for review

mqole commented:
**WE WILL GET THESE TESTS TO SUCCEED OR DIE TRYING!!!!!!!!!!!!!!**

mqole commented:
oh thats like Scary scary.

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/Baked/pie.yml:
```diff
@@ -362,9 +363,9 @@
-          Quantity: 6.25
+          Quantity: 11.25
+        - ReagentId: Vitamin
+          Quantity: 3.75
```
it might be angry because this adds uo to exactly 15?

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/Baked/pie.yml:
```diff
@@ -743,11 +745,11 @@
-        - ReagentId: JuiceApple
-          Quantity: 20
+        - ReagentId: Vitamin
+          Quantity: 8
```
maxvol 30
reagents 68

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/Baked/pizza.yml:
```diff
@@ -661,20 +659,18 @@
-          Quantity: 8
-        - ReagentId: JuiceTomato # imp
-          Quantity: 20
+          Quantity: 14
```
maxvol 85
reagents 91

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/burger.yml:
```diff
@@ -932,19 +933,18 @@
+        - ReagentId: Vitamin
+          Quantity: 3 # imp
         - ReagentId: TableSalt
           Quantity: 5
```
maxvol: 60
reagents: 60

even if this isnt making it angry i do think there should be a small maxvol buffer for poisoning shenanigans

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/meals.yml:
```diff
@@ -669,33 +668,28 @@
-        - ReagentId: JuiceOrange # imp
-          Quantity: 10
         - ReagentId: Desoxyephedrine # imp
           Quantity: 10
```
maxvol: 90
reagents: 90

see above

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/noodles.yml:
```diff
@@ -56,12 +56,12 @@
-          Quantity: 20
+          Quantity: 39
+        - ReagentId: Vitamin
+          Quantity: 6
```
maxvol: 45
reagents: 45

see above

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/soup.yml:
```diff
@@ -307,15 +308,9 @@
-          Quantity: 10
-        - ReagentId: JuiceGrape # imp
-          Quantity: 10
+          Quantity: 16 # imp
```
maxvol: 40
reagents: 40

see above

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/soup.yml:
```diff
@@ -842,17 +831,15 @@
         #- ReagentId: Omnizine # imp
         #  Quantity: 5
         - ReagentId: Histamine
           Quantity: 3 # imp
```
maxvol: 40
reagents: 48

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/soup.yml:
```diff
@@ -1054,15 +1038,13 @@
         - ReagentId: Water # imp
           Quantity: 10
         - ReagentId: JuiceCarrot # imp
           Quantity: 10
```
maxvol: 55
reagents: 59

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/soup.yml:
```diff
@@ -1272,25 +1252,24 @@
-        - ReagentId: JuiceTomato # imp
-          Quantity: 10
         - ReagentId: CarpoToxin # imp
           Quantity: 5
```
maxvol: 30
reagents: 30

see above

mqole commented on Resources/Prototypes/Entities/Objects/Consumable/Food/soup.yml:
```diff
@@ -1338,14 +1318,14 @@
           Quantity: 5
-        - ReagentId: JuiceBungo
+        - ReagentId: Enzyme # imp
           Quantity: 10
```
maxvol: 35
reagents: 43

mqole reviewed (commented):
think this might be whats causing the test fail... only looked at changes from the commit where test broke, so you might have gotten some already, in which case, ignore.

general things are that some items have a greater quantity of reagents than whats expected in their max volume, and some others have a quantity of Exactly max volume (which makes it impossible to add flavouring, or poison)

mqole reviewed (commented):
few more from newest commits

hivehum commented:
hero to me... i suspected it was the same issue. going to do another pass where i make sure everything has a bit of extra room for fun and poison

hivehum commented:
added milk back into everything that used it as an ingredient (niche interaction with skeletons + would be funny if we get lactose intolerance at some point). also added EggCooked to everything that had egg in it. golden apple streusel tart now uses golden apples like it appears to have initally been intended

mqole commented:
jesus christ

mqole reviewed (approved):
WE WIN!
